min.py

```diff
@@ -135,6 +136,36 @@ def get_min_scipy_version(min_python, min_numpy):
     return df.SciPy_version
 
 
+def find_pkg_mismatches(pkg_name, pkg_version, fnames):
+    """
+    Determine if any package version has mismatches
+    """
+    pkg_mismatches = []
+
+    for fname in fnames:
+        with open(fname, "r") as file:
+            for line_num, line in enumerate(file, start=1):
+                l = line.strip().replace(" ", "").lower()
+                matches = re.search(
+                    rf"""
+                        {pkg_name}  # Package name
+                        [=><:"\'\[\]]+  # Zero or more special characters
+                        (\d+\.\d+[\.0-9]*)  # Capture "version" in `matches`
+                        """,
+                    l,
+                    re.VERBOSE,  # Ignores all whitespace in pattern
+                )
+                if matches is not None:
+                    version = matches.groups()[0]
+                    if version != pkg_version:
+                        pkg_mismatches.append(
+                            f'Package Mismatch Found: "{pkg_name}" "{version}" '
+                            f"in {fname}:{line_num}"
+                        )
+
+    return "\n".join(pkg_mismatches)
```

Comment:
Maybe it's just me but I don't like doing the join here and I'd rather return a list of strings rather than a single long string. Perhaps it's because it keeps this function flexible.

I know that I'm being picky and possibly annoying now but maybe a list of strings is too rigid and, instead, we should do:

```
pkg_mistaches.append((pkg_name, version, fname, line_num))
```

And then all of the print formatting happens in `__main__`. I think I like that better because it'll likely be easier to maintain/change in the future. And "FUTURE Sean/Joey will thank PAST Sean/Joey" for not painting ourselves into a corner

Reply:
@seanlaw

No annoyance here - it's extremely helpful for me to see and understand your design decisions, so again - I very much appreciate the feedback! I've fixed the best I can according to this approach!

Let me know if I need to update further!